common/building/src/utils.rs

```diff
@@ -0,0 +1,40 @@
+// Copyright 2022 Datafuse Labs.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use std::fmt::Display;
+use std::fmt::Error;
+use std::fmt::Formatter;
+
+pub struct FormatVec(pub Vec<String>);
+
+impl Display for FormatVec {
+    fn fmt(&self, f: &mut Formatter) -> Result<(), Error> {
+        let mut comma_separated = String::new();
+
+        let v = &self.0;
+
+        if v.is_empty() {
+            return Ok(());
+        }
+
+        for s in &self.0[0..v.len() - 1] {
+            comma_separated.push_str(s);
+            comma_separated.push_str(", ");
+        }
+
+        comma_separated.push_str(&self.0[v.len() - 1]);
```

Comment:
How about using `self.0.iter().join(", ")`?

Reply:
Makes sense. I just wanted a pretty print look at first. Let me try this.